### DISCUSSION TITLE
configuring Argo Workflows to use containerd

### DISCUSSION THREAD

**Discussion by chrisadkin** [argoproj/argo-workflows#8609] [Q&A]
I'm using version 3.3 of Argo Workflows with Kubernetes 1.22.5 using containerd, my understanding is that if the containersRuntimeExecutor in the workflow-controller-configmap is set to include pns, thus:

```
apiVersion: v1
data:
  containerRuntimeExecutors: |
    - name: pns
      selector:
        matchLabels:
          workflows.argoproj.io/container-runtime-executor: pns
    - name: emissary
      selector:
        matchLabels:
          workflows.argoproj.io/container-runtime-executor: emissary
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ConfigMap","metadata":{"annotations":{},"name":"workflow-controller-configmap","namespace":"argo"}}
  creationTimestamp: "2022-05-04T12:26:30Z"
  name: workflow-controller-configmap
  namespace: argo
  resourceVersion: "14979838"
  uid: a1e5474d-1a9b-4d6e-bf8a-f1b6f7860c89
```

Argo Workflows workflows should work quite happily with containerd, however, despite this I am getting:

```
Events:
  Type     Reason       Age                From               Message
  ----     ------       ----               ----               -------
  Normal   Scheduled    49s                default-scheduler  Successfully assigned default/hello-world-99rh8 to z-ca-arc-compute1
  Warning  FailedMount  18s (x7 over 49s)  kubelet            MountVolume.SetUp failed for volume "docker-sock" : hostPath type check failed: /var/run/docker.sock is not a socket file
```

I suspect there is something I'm missing somewhere in terms of configuration and that this is not a bug, could someone please point me in the right direction for getting this fixed.

**Comment by terrytangyuan** [comment]
Try switching to emissary executor which should be the default. Other executors are deprecated.

**Reply by chrisadkin** [reply]
Done that, its made no difference :-(

**Reply by terrytangyuan** [reply]
Can you try removing `pns` from `containerRuntimeExecutors`?

**Reply by chrisadkin** [reply]
Still made no difference.

**Reply by terrytangyuan** [reply]
Could you share the controller log?

**Reply by chrisadkin** [reply]
Is there anything I need to do in order to get the workflow engine to pick up the configmap changes ?

**Reply by terrytangyuan** [reply]
You should be able to see which executor it is using in controller log. Try re-deploying your controller.

**Reply by chrisadkin** [reply]
Controller logs attached as requested:

[workflow-controller-64c947c746-xhdn6-logs.txt](https://github.com/argoproj/argo-workflows/files/8621717/workflow-controller-64c947c746-xhdn6-logs.txt)

**Reply by terrytangyuan** [reply]
I still see `workflows.argoproj.io/container-runtime-executor: pns` in the log. You should remove it from your configmap and then restart/redeploy your controller.

**Reply by chrisadkin** [reply]
Based on what is in the log the last run is using emissary if I'm reading it correctly

**Reply by terrytangyuan** [reply]
Can you restart your controller?

**Comment by natalytvinova** [comment]
Hi @terrytangyuan  I'm having the same error trying to use containerd with Argo 3.4.1. I have already tried using containerRuntimeExecutors Emissary and Pns but the error still appears. Here are my controller logs:
[controller-logs.txt](https://github.com/argoproj/argo-workflows/files/9714524/controller-logs.txt)
